### DISCUSSION TITLE
Discord /invite command

### DISCUSSION THREAD

**Discussion by VioletLeporid** [hrfee/jfa-go#294] [Ideas]
The current Discord integration is great, but one QoL feature that could be added is an admin-only command for sending invites. It could be set to only accept the command from the discord account that is linked to the Jellyfin Admin account, and allow generating and sending invite links directly through discord rather than using the web interface. An example command might look like this:

`/invite <username> <profile> <label> <expiration>`

with only the username field being required. This feature would allow interacting with jfa-go exclusively through discord after initial setup.

**Comment by VioletLeporid** [comment]
I decided to take a crack at implementing this myself, with my absolute lack of experience with both Go and the Discord API. I think I eventually managed to figure it out, but I'm getting this error:
`Failed to construct invite message: template: pattern matches no files: invite-email.html`
This seems to have nothing at all to do with the changes I made so I'm wondering if the source hosted here is the same as the compiled version, which I was able to run with no problems. I might look further into it later but seems to just be an issue constructing the message to send.

Edit: Should've also mentioned the built-in invite feature doesn't seem to work either. Maybe I made a mistake compiling it?

**Comment by VioletLeporid** [comment]
I believe this should be functional if compiled correctly (not sure what I'm doing wrong there but I don't have a Linux environment atm so it could just be a problem related to me using Windows)

For now I have just one option for the user to be invited, but I commented some [hopefully functional] code for the other options to potentially add in the future.

Edit: I just added in the check for admin and it appears to be working, but my test environment is a discord server with only two users, so might be worth some further testing or some looking over by someone who understands the code better, but I'm pretty sure I did it right. I just need to add the response messages in discord and then see if I can figure out if I can get it to compile properly and run the whole thing.

Also small QoL change - I added a status message, since that's fairly standard practice for Discord bots now. It will display a status of "Playing /start" using the config for start_command

https://github.com/VioletLeporid/jfa-go/blob/main/discord.go

**Comment by VioletLeporid** [comment]
Okay I spun up a VM and I'm also having issues compiling on Ubuntu, here's the error I get from using `make DEBUG=on all`

```
*/jfa-go/node_modules/typescript/lib/tsc.js:93
   for (let i = startIndex ?? 0; i < array.length; i++) {
                            ^

SyntaxError: Unexpected token '?'
```

followed by several references but the main one is `jfa-go/nodemofules/typescript/bin/tsc:2:1`

And this is what I get from `./scripts/version.sh goreleaser build --snapshot --clean`

```
error=hook failed: npx tailwindcss -i data/web/css/bundle.css -o data/bundle.css --content "html/crash.html": exit status 1; output: */jfa-go/node_modules/yaml/dist/compose/composer.js:33
          if (prelude[i + 1]?.[0] !== '#')
                             ^

SyntaxError: Unexpected token '.'
```  
followed again by several references, most of which are `internal/modules/cjs/loader.js` one at `internal/modules/cjs/helpers.js:74:18` and one at `jfa-go/node_modules/yaml/dist/index.js:3:16`

I'm not sure what I'm doing wrong here. I can download the release binary and it runs fine, but I seem to be unable to make one myself using either method on Windows or Ubuntu

**Reply by hrfee** [reply]
Those errors seem to just be due to an outdated nodejs, attempting to build with the nodejs in the ubuntu repo fails, and warnings popup saying at least v14 is required. Try getting a newer version [from here](https://github.com/nodesource/distributions#installation-instructions).

**Reply by hrfee** [reply]
Also, make sure Go's bin directory is in your path.

**Reply by hrfee** [reply]
Last note, Go version in ubuntu repositories is also outdated, 1.20 at least is needed. A PPA can be found [here](https://github.com/golang/go/wiki/Ubuntu)

**Reply by VioletLeporid** [reply]
Thanks! Building with Goreleaser worked perfectly after I updated nodejs, now I can properly test my changes

**Comment by hrfee** [comment]
~~I'm making some changes locally so that this can be merged, would you be able to create a Pull request for the fork so I can commit to it? Thanks.~~

~~Found out i can do this myself, created one: #298~~ Nevermind, doing this means I can't push to the fork, so would you be able to do this, sorry.

**Reply by VioletLeporid** [reply]
Okay I created the request. I haven't added all of the finishing touches yet like response messages, so I guess that'll be whoever gets to it first.

**Reply by VioletLeporid** [reply]
Actually maybe I should just leave the response messages since I don't want to mess with the lang stuff

**Reply by hrfee** [reply]
No problem, I've just pushed my tweaks which include a response.

**Reply by VioletLeporid** [reply]
Thanks for indulging me, this was a great opportunity for me to learn some new stuff. There are a couple other discord features that I was thinking would be nice, so now that I somewhat know what I'm doing, I'll probably mess around with it some more and see what I can do.

**Reply by hrfee** [reply]
No problem and thank you, it's always great to have others contribute code, and more features for the Discord bot is something that's been requested many times before.
